core/node/notifications/auth.go

```diff
@@ -36,15 +35,16 @@ type (
 )
 
 func (c authenticationChallenge) Verify(
+	ctx context.Context,
 	challenge [challengeLength]byte,
 	signature []byte,
 	delegateSig []byte,
 	delegateExpiryEpochMs int64,
-) bool {
+) error {
 	// ensure that the auth challenge nor the delegateExpiryEpoch hasn't expired.
 	now := time.Now()
-	if now.After(c.expires) { // ignored for now || now.After(time.Unix(delegateExpiryEpochMs/1000, 0)) {
-		return false // expired
+	if now.After(c.expires) || (len(delegateSig) > 0 && now.After(time.Unix(delegateExpiryEpochMs/1000, 0))) {
+		return RiverError(Err_UNAUTHENTICATED, "authentication expired").Tag("expires", c.expires).Tag("delegateExpiryEpochMs", delegateExpiryEpochMs)
```

Comment:
Nitpick, RiverError accepts key/value pairs as tags. No need to chain Tag calls.

Reply:
i was just following the convention in the file. I can replace them all.